### ISSUE TITLE
Wrong detected error "Possibly uninitialized variable"

### ISSUE THREAD

**Issue by webdesus** [intellij-rust/intellij-rust#4008]
## Environment

* **Intellij-Rust plugin v0.2.99.2126-191:**
* **Rust toolchain rustup 1.16.0 (beab5ac2b 2018-12-06)**
* **IDE name and clion 2019.1:**
* **Operating system: linux**

## Problem description
Wrong detected error "Possibly uninitialized variable". Plugin not consider cases: "unimplemented", "unreachable", etc.

## Steps to reproduce

```rust
    let value;
    match &enum_item {
        ItemType::Element1(f_data) => {
            value = 123;
        },
        ItemType::Element2(_,f_data)=>{

            value = 23;
        }
        _ => unimplemented!(),
    }
    return value  // Use of posibly uninitialized variable  
```
